### PR TITLE
new_owner: a owner implement, the owner is a manager of changefeeds

### PR DESCRIPTION
**Pull request by leoppro**
### What problem does this PR solve? 
a owner implement, the owner is a manager of changefeeds

the owner implements the Reactor interface, Etcd worker driver the owner tick by tick.
the owner's responsible for starting/stopping the changefeed and send admin job to the changefeed

### Check List 

Tests 

 - Unit test
 - Integration test

### Release note

- No release note

### CODE REVIEW

File: cdc/owner/owner.go [hunk 1]
```diff
@@ -0,0 +1,271 @@
...
+			return nil, workload != nil, nil
+		})
+	}
+}
+
+func (o *Owner) updateMetrics(state *model.GlobalReactorState) {
+	// Keep the value of prometheus expression `rate(counter)` = 1
+	// Please also change alert rule in ticdc.rules.yml when change the expression value.
+	now := time.Now()
+	ownershipCounter.Add(float64(now.Sub(o.lastTickTime) / time.Second))
```
Comment (amyangfei) [inline cdc/owner/owner.go:219]: Seems this fix a bug in old owner 👍

File: cdc/owner/owner.go [hunk 2]
```diff
@@ -0,0 +1,271 @@
...
+		}
+		cfReactor.Tick(ctx, changefeedState, state.Captures)
+	}
+	if len(o.changefeeds) != len(state.Changefeeds) {
+		for changefeedID, cfReactor := range o.changefeeds {
+			if _, exist := state.Changefeeds[changefeedID]; exist {
+				continue
+			}
+			cfReactor.Close()
+			delete(o.changefeeds, changefeedID)
```
Comment (amyangfei) [inline cdc/owner/owner.go:135]: the `cfReactor` is not used afterward, I notice there exist some operations in `func (c *changefeed) releaseResources()`, such as
```
c.metricsChangefeedCheckpointTsGauge = nil
c.metricsChangefeedCheckpointTsLagGauge = nil
c.initialized = false  
```
are these useless

File: cdc/owner/owner.go [hunk 3]
```diff
@@ -0,0 +1,271 @@
...
+type Owner struct {
+	changefeeds map[model.ChangeFeedID]*changefeed
+
+	gcManager *gcManager
+
+	ownerJobQueue   []*ownerJob
+	ownerJobQueueMu sync.Mutex
+	lastTickTime    time.Time
+
+	close int32
```
Comment (amyangfei) [inline cdc/owner/owner.go:72]: `closed` is better, `close` is a verb and it is also a keyword

File: cdc/owner/owner.go (hunk 1 above)
Comment (leoppro) [reply cdc/owner/owner.go:219]: ok

File: cdc/owner/owner.go [hunk 4]
```diff
@@ -0,0 +1,271 @@
...
+}
+
+func (o *Owner) updateMetrics(state *model.GlobalReactorState) {
+	// Keep the value of prometheus expression `rate(counter)` = 1
+	// Please also change alert rule in ticdc.rules.yml when change the expression value.
+	now := time.Now()
+	ownershipCounter.Add(float64(now.Sub(o.lastTickTime) / time.Second))
+	o.lastTickTime = now
+
+	ownerMaintainTableNumGauge.Reset()
```
[the remark this answers is not in the document]
Comment (leoppro) [reply cdc/owner/owner.go:222]: Theoretically, it is possible that we need to maintain the metadata of the last tick and compare it with the current tick

**Review by liuzix — APPROVED** [review]

**Pull request merged**